src/main/java/com/gmail/llmdlio/townyflight/listeners/EnemyEnterTownListener.java

```diff
@@ -0,0 +1,177 @@
+package com.gmail.llmdlio.townyflight.listeners;
+
+import com.gmail.llmdlio.townyflight.TownyFlight;
+import com.gmail.llmdlio.townyflight.TownyFlightAPI;
+import com.gmail.llmdlio.townyflight.config.Settings;
+import com.palmergames.bukkit.towny.TownyAPI;
+import com.palmergames.bukkit.towny.event.player.PlayerEntersIntoTownBorderEvent;
+import com.palmergames.bukkit.towny.event.player.PlayerExitsFromTownBorderEvent;
+import com.palmergames.bukkit.towny.object.Resident;
+import com.palmergames.bukkit.towny.object.Town;
+import com.palmergames.bukkit.towny.utils.CombatUtil;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerJoinEvent;
+import org.bukkit.event.player.PlayerQuitEvent;
+import org.bukkit.event.player.PlayerTeleportEvent;
+
+public class EnemyEnterTownListener implements Listener {
```

Comment:
Rename to something more descriptive, ie OutsidersInTownListener.

Reply:
All of the other methods should lose the Enemy naming as well, since this is not limited to just enemies.